src/rgw/rgw_admin_common.h

```diff
@@ -0,0 +1,291 @@
+#ifndef CEPH_RGW_ADMIN_COMMON_H
+#define CEPH_RGW_ADMIN_COMMON_H
+
+#include "cls/rgw/cls_rgw_types.h"
+
+#include "common/ceph_json.h"
+#include <common/safe_io.h>
+
+#include "rgw_common.h"
+#include "rgw_rados.h"
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_rgw
+
+enum RgwAdminCommandGroup {
+  INVALID = -1,
+  BI,
+  BILOG,
+  BUCKET,
+  BUCKET_SYNC,
+  OBJECT,
+  POLICY,
+  RESHARD,
+  DATA_SYNC,
+  METADATA_SYNC,
+  PERIOD,
+  REALM,
+  ZONE,
+  ZONE_PLACEMENT,
+  ZONEGROUP,
+  ZONEGROUP_PLACEMENT,
+  GLOBAL_QUOTA,
+  QUOTA,
+  ROLE,
+  ROLE_POLICY,
+  USER,
+  SUBUSER,
+  CAPS,
+  KEY,
+  DATALOG,
+  LOG,
+  MDLOG,
+  REPLICALOG,
+  GC,
+  LC,
+  METADATA,
+  OLH,
+  OPSTATE,
+  ORPHANS,
+  POOL,
+  SYNC_ERROR,
+  SYNC_STATUS,
+  USAGE,
+  USER_LIST,
```

Comment:
should be part of user

Reply:
I would say user_list is very different from all other user commands (they all operate with user_op; to the contrary, this command has the same logic as metadata_list), so I would like to keep it separate.